src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -492,16 +538,18 @@ function copy_to_coupled_boundary!(boundary_condition::BoundaryConditionCoupled{
     for cell in cells
         i_node = i_node_start
         j_node = j_node_start
-
-        for i in eachnode(solver)
-            for v in 1:size(u, 1)
-                boundary_condition.u_boundary[v, i, cell] = u[v, i_node, j_node,
-                                                              linear_indices[i_cell,
-                                                                             j_cell]]
-            end
+        element_id = linear_indices[i_cell, j_cell]
+
+        for element_id in eachnode(solver)
+            x = @view cache.elements.node_coordinates[:, i_node, j_node,
+                                                      linear_indices[i_cell, j_cell]]
+            u_node = u[:, i_node, j_node, linear_indices[i_cell, j_cell]]
+            converted_u = coupling_converter(x, u_node)
+            u_boundary[:, element_id, cell] = converted_u
```

Comment:
```suggestion
            @inbounds for i in eachindex(converted_u)
                u_boundary[i, element_id, cell] = converted_u[i]
            end
```
This is probably faster (even without `@inbounds` because Julia).